notebooks/python/neo4j/README.md

```diff
@@ -0,0 +1,253 @@
+
+We know that Spark can collect results executed on many processes in parallel.  So how does Spark parallelize this query:
+
+    MATCH (p:Person) RETURN p
+
+In fact, the Spark Connector parses and rewrites the query.  It first figures out how many rows your query will return (let’s say 100) and then rewrites the query so it can be sharded, this way:
+
+    MATCH (p:Person) RETURN p SKIP 0 LIMIT 20
+    MATCH (p:Person) RETURN p SKIP 20 LIMIT 20
+    MATCH (p:Person) RETURN p SKIP 40 LIMIT 20
+    MATCH (p:Person) RETURN p SKIP 60 LIMIT 20
+    MATCH (p:Person) RETURN p SKIP 80 LIMIT 20
+
+And this is why you’ll have a hard time using SKIP and LIMIT in your Cypher queries, when running under the Spark Connector!
+
+## Reading from and Writing to a Database
+In the original blog, Nathan loads data using a [`LOAD CSV`](https://neo4j.com/docs/cypher-manual/current/clauses/load-csv/) Cypher statement which elegantly handles downloading CSV text files from a cloud source, batch loading, and flexible casting through a Neo4j unwind statement.
+
+In many cases, the system of record will be an RDBMS like BigQuery.  When using either of these databases, their respective Spark Connector and JDBC libraries must be loaded into the Spark session.
+
+At the time of writing, DataProc supported Spark 2.4 and the following packages are required for BigQuery.
+
+    .config('spark.jars.packages',
+    "org.neo4j:neo4j-connector-apache-spark_2.12:4.1.3_for_spark_2.4,
+    com.google.cloud.spark:spark-bigquery-with-dependencies_2.12:0.26.0")
+
+Now that the Spark session is initialized with the right libraries (the full initializer is in the code), we can read and write to our databases.
+
+## Data Types
+When importing data from a relational database, pay attention to data types.  
+
+For example, some databases do not support float data types so it is necessary to downcast all decimals and doubles to floats, after querying.
+
+You can inspect data types with the `df.printSchema()` method.
+
+There are many possible ways to recast columns in a DataFrame; here `selectExpr` is shown:
+
+    floatCastSelectExpr=["NAME as name","cast(POPULATION as long) population",\
+    "cast(PERCENTOVER25WITHBACHELORS as float) percentOver25WithBachelors",\
+    "cast(MEDIANHOMEPRICE as int) medianHomePrice",\
+    "cast(MEDIANHOUSEHOLDINCOME as int) medianHouseholdIncome"]
+
+    castedDf = rawDf.selectExpr(floatCastSelectExpr)
+
+After down-casting doubles and decimals which Neo4j does not support, to floats or integers which it does, populating Neo4j nodes from a Spark DataFrame is trivial, and identical regardless of source.
+
+    (castedDf.write
```

Comment:
Please surround with code block, trailing back-slashes are needed for multi-line statement.

Also - why is the statement surrounded with parenthesis?